src/linear_algebra/unitary_group.lean

```diff
@@ -0,0 +1,191 @@
+
+@[ext] lemma ext (A B : unitary_group n) : (∀ i j, A i j = B i j) → A = B :=
+(unitary_group.ext_iff A B).mpr
+
+instance : has_inv (unitary_group n) :=
+⟨λ A, ⟨A.1†, matrix.nonsing_inv_left_right _ _ $
+  by { dsimp only, rw [conj_transpose_conj_transpose, A.2]}⟩⟩
+
+noncomputable instance : has_mul (unitary_group n) :=
+⟨λ A B, ⟨matrix.mul A.1  B.1, by rw [conj_transpose_mul, ←matrix.mul_assoc,
+                                     matrix.mul_assoc _ _ A.1, A.2, matrix.mul_one, B.2],⟩⟩
+
+instance : has_one (unitary_group n) :=
+⟨⟨1, by rw [conj_transpose_one, matrix.one_mul]⟩⟩
+
+instance : inhabited (unitary_group n) := ⟨1⟩
+
+section coe_lemmas
+
+variables (A B : unitary_group n)
+
+@[simp] lemma inv_val : ↑(A⁻¹) = A† := rfl
+
+@[simp] lemma inv_apply : ⇑(A⁻¹) = A† := rfl
+
+@[simp] lemma mul_val : ↑(A * B) = A ⬝ B := rfl
+
+@[simp] lemma mul_apply : ⇑(A * B) = (A ⬝ B) := rfl
+
+@[simp] lemma one_val : ↑(1 : unitary_group n) = (1 : matrix n n ℂ) := rfl
+
+@[simp] lemma one_apply : ⇑(1 : unitary_group n) = (1 : matrix n n ℂ) := rfl
+
+@[simp] lemma to_lin'_mul :
+  to_lin' (A * B) = (to_lin' A).comp (to_lin' B) :=
+matrix.to_lin'_mul A B
+
+@[simp] lemma to_lin'_one :
+  to_lin' (1 : unitary_group n) = linear_map.id :=
+matrix.to_lin'_one
+
+end coe_lemmas
+
+noncomputable instance : group (unitary_group n) :=
+{ mul_assoc := λ A B C, subtype.eq (A.val.mul_assoc B.val C.val),
+  one_mul := λ M, subtype.eq $ one_mul _,
+  mul_one := λ M, subtype.eq $ mul_one _,
+  mul_left_inv := λ A, subtype.eq A.2,
+  ..unitary_group.has_mul,
+  ..unitary_group.has_inv,
+  ..unitary_group.has_one }
+
+/-- `to_linear_equiv A` is matrix multiplication of vectors by `A`, as a linear equivalence. -/
+noncomputable def to_linear_equiv (A : unitary_group n) : (n → ℂ) ≃ₗ[ℂ] (n → ℂ) :=
+{ inv_fun := A⁻¹.to_lin',
+  left_inv := λ x, calc
+    A⁻¹.to_lin'.comp A.to_lin' x
+        = (A⁻¹ * A).to_lin' x : by rw [←to_lin'_mul]
+    ... = x : by rw [mul_left_inv, to_lin'_one, id_apply],
+  right_inv := λ x, calc
+    A.to_lin'.comp A⁻¹.to_lin' x
+        = (A * A⁻¹).to_lin' x : by rw [←to_lin'_mul]
+    ... = x : by rw [mul_right_inv, to_lin'_one, id_apply],
+  ..matrix.to_lin' A }
+
+/-- `to_GL` is the map from the unitary group to the general linear group -/
+noncomputable def to_GL (A : unitary_group n) : general_linear_group ℂ (n → ℂ) :=
+general_linear_group.of_linear_equiv (to_linear_equiv A)
+
+lemma coe_to_GL (A : unitary_group n) :
+  (@coe (units _) _ _ (to_GL A)) = A.to_lin' :=
+rfl
+
+@[simp]
+lemma to_GL_one : to_GL (1 : unitary_group n) = 1 :=
+by { ext1 v i, rw [coe_to_GL, to_lin'_one], refl }
+
+@[simp]
+lemma to_GL_mul (A B : unitary_group n) :
+  to_GL (A * B) = to_GL A * to_GL B :=
+by { ext1 v i, rw [coe_to_GL, to_lin'_mul], refl }
+
+/-- `unitary_group.embedding_GL` is the embedding from `unitary_group n`
+  to `general_linear_group n ℂ`. -/
+noncomputable def embedding_GL : (unitary_group n) →* (general_linear_group ℂ (n → ℂ)) :=
```

Comment:
I'd maybe suggest `to_Gl_hom` instead (and to add a `@[simps]` attribute). `embedding` suggests to me that you're going to prove its injective, which you don't.

Reply:
I've copied this, and in fact most of the definitions/theorems, from `special_linear_group`. I'm not 100% sure why it's called `embedding` there.